pkg/e2e/infra/utils.go

```diff
@@ -63,3 +69,112 @@ func isOneMachinePerNode(client runtimeclient.Client) bool {
 	}
 	return true
 }
+
+// getClusterSize returns the number of nodes of the cluster
+func getClusterSize(client runtimeclient.Client) (int, error) {
+	nodeList := corev1.NodeList{}
+	var size int
```

Comment:
I don't think you need the temporary. At the end of the function you can just `return len(nodeList.Items), nil`?